docs/root/start/sandboxes/route-mirror.rst

```diff
@@ -0,0 +1,213 @@
+both the ``service1`` and ``service1-mirror`` services got the request made
+in Step 2.
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/route-mirror
+
+   $ docker-compose logs service1
+   ...
+   Host: localhost:10000
+   192.168.80.6 - - [06/Oct/2022 03:56:22] "GET /service/1 HTTP/1.1" 200 -
+
+   $ docker-compose logs service1-mirror
+   ...
+   Host: localhost-shadow:10000
+   192.168.80.6 - - [06/Oct/2022 03:56:22] "GET /service/1 HTTP/1.1" 200 -
+
+You can also see that for the request to the ``service1-mirror`` service, the
+``Host`` header was modified by Envoy to have a ``-shadow`` suffix in the
+hostname.
+
+Step 4: Make a request to the route mirrored by request header
+**************************************************************
+
+In this step, we will see a demonstration where the request specifies via a header, ``x-mirror-cluster``,
+the cluster that envoy will mirror the request to.
+
+Let's send a request to the ``envoy-front-proxy`` service which forwards the request to
+``service2`` and also mirrors the request to the cluster named, ``service2-mirror``.
+
+.. code-block:: console
+
+  $ curl --header "x-mirror-cluster: service2-mirror" localhost:10000/service/2
+  Hello from behind Envoy (service 2)!
+
+
+Step 5: View logs for the request mirrored by request header
+************************************************************
+
+The logs show that both the ``service2`` and ``service2-mirror`` services
+got the request.
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/route-mirror
+
+   $ docker-compose logs service2
+   ...
+   Host: localhost:10000
+   192.168.80.6 - - [06/Oct/2022 03:56:22] "GET /service/2 HTTP/1.1" 200 -
+
+   $ docker-compose logs service2-mirror
+   ...
+   Host: localhost-shadow:10000
+   192.168.80.6 - - [06/Oct/2022 03:56:22] "GET /service/2 HTTP/1.1" 200 -
+
+You can also see that for the request to the ``service2-mirror`` service, the
+``Host`` header was modified by Envoy to have a ``-shadow`` suffix in the
+hostname.
+
+Step 6: Missing or invalid cluster name in request header
+*********************************************************
+
+If you do not specify the ``x-mirror-cluster`` in the request to ``service2``,
+a request to ``service1`` will not automatically be mirrored to ``service2-mirror``.
+
+.. code-block:: console
+
+  $ curl localhost:10000/service/2
+  Hello from behind Envoy (service 2)!
+
+View the logs for the ``service2`` and ``service2-mirror`` services.
+
+.. code-block:: console
+
+   $ pwd
+   envoy/examples/route-mirror
+
+   $ docker-compose logs service2
+   ...
+   Host: localhost:10000
+   192.168.80.6 - - [06/Oct/2022 03:56:22] "GET /service/2 HTTP/1.1" 200 -
+
+   $ docker-compose logs service2-mirror
+   # No new logs
+
+Similarly, if we specify a value for the ``x-mirror-cluster`` header such that there
+is no such cluster defined with that name, ``service2-mirror`` will not receive
+the request:
```

Comment:
i think collapse these so that you do all of the requests in the first code block and then you can separate out checking the logs to a separate step